src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -1386,61 +1387,53 @@ void GcsActorManager::Initialize(const GcsInitData &gcs_init_data) {
+        actor_table_data, actor_task_spec, actor_state_counter_);
+
+    registered_actors_.emplace(actor_id, actor);
+    function_manager_.AddJobReference(actor->GetActorID().JobId());
+    if (!actor->GetName().empty()) {
+      named_actors_[actor->GetRayNamespace()][actor->GetName()] = actor->GetActorID();
+    }
 
-      if (actor_table_data.state() == ray::rpc::ActorTableData::DEPENDENCIES_UNREADY) {
-        const auto &owner = actor->GetOwnerAddress();
-        const auto &owner_node = NodeID::FromBinary(owner.raylet_id());
-        const auto &owner_worker = WorkerID::FromBinary(owner.worker_id());
-        RAY_CHECK(unresolved_actors_[owner_node][owner_worker]
-                      .emplace(actor->GetActorID())
-                      .second);
-      } else if (actor_table_data.state() == ray::rpc::ActorTableData::ALIVE) {
-        created_actors_[actor->GetNodeID()].emplace(actor->GetWorkerID(),
-                                                    actor->GetActorID());
-      }
+    if (actor_table_data.state() == ray::rpc::ActorTableData::DEPENDENCIES_UNREADY) {
+      const auto &owner = actor->GetOwnerAddress();
+      const auto &owner_node = NodeID::FromBinary(owner.raylet_id());
+      const auto &owner_worker = WorkerID::FromBinary(owner.worker_id());
+      RAY_CHECK(unresolved_actors_[owner_node][owner_worker]
+                    .emplace(actor->GetActorID())
+                    .second);
+    } else if (actor_table_data.state() == ray::rpc::ActorTableData::ALIVE) {
+      created_actors_[actor->GetNodeID()].emplace(actor->GetWorkerID(),
+                                                  actor->GetActorID());
+    }
 
-      if (!actor->IsDetached()) {
-        // This actor is owned. Send a long polling request to the actor's
-        // owner to determine when the actor should be removed.
-        PollOwnerForActorOutOfScope(actor);
-      }
+    if (!actor->IsDetached()) {
+      // This actor is owned. Send a long polling request to the actor's
+      // owner to determine when the actor should be removed.
+      PollOwnerForActorOutOfScope(actor);
+    }
 
-      if (!actor->GetWorkerID().IsNil()) {
-        RAY_CHECK(!actor->GetNodeID().IsNil());
-        node_to_workers[actor->GetNodeID()].emplace_back(actor->GetWorkerID());
-      }
-    } else {
-      dead_actors.push_back(actor_id);
-      auto actor = std::make_shared<GcsActor>(actor_table_data, actor_state_counter_);
-      destroyed_actors_.emplace(actor_id, actor);
-      sorted_destroyed_actor_list_.emplace_back(actor_id,
-                                                (int64_t)actor_table_data.timestamp());
+    if (!actor->GetWorkerID().IsNil()) {
+      RAY_CHECK(!actor->GetNodeID().IsNil());
+      node_to_workers[actor->GetNodeID()].emplace_back(actor->GetWorkerID());
     }
   }
-  if (!dead_actors.empty()) {
-    RAY_CHECK_OK(
-        gcs_table_storage_->ActorTaskSpecTable().BatchDelete(dead_actors, nullptr));
```

Comment:
We still need this

Reply:
It’s covered by AddDestroyedActorToCache